src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java

```diff
@@ -93,19 +96,20 @@ protected void loggedIn(@Nonnull String username) {
      *
      * @param username name or ID of the user who logged out.
      */
-     @Override
-     protected void loggedOut(@Nonnull String username) {
-         String currentTime = Instant.now().toString();
-         Logout logout = LogEventFactory.getEvent(Logout.class);
+    @Override
+    protected void loggedOut(@Nonnull String username) {
+        String currentTime = Instant.now().toString();
+        Logout logout = LogEventFactory.getEvent(Logout.class);
 
-         RequestContext.setNodeName("master");
-         RequestContext.getRequestId();
-         RequestContext.setIpAddress(Stapler.getCurrentRequest().getRemoteAddr());
-         logout.setUserId(username);
-         logout.setTimestamp(currentTime);
-         logout.logEvent();
-         RequestContext.clear();
-     }
+        RequestContext.setNodeName("master");
+        RequestContext.getRequestId();
+        RequestContext.setIpAddress(Stapler.getCurrentRequest().getRemoteAddr());
```

Comment:
This should be done via the servlet filter. You could leave this line here with a TODO comment mentioning the servlet filter task.

Reply:
as discussed earlier.. will refactor `UserLogListener` and move those `RequestContext` methods out.